File: MutableAI/Java/Q7.java
package Java;

public class Q7 {
    public int countPrimes(int n) {
        //Given an integer n, return the number of prime numbers 
        //that are strictly less than n. with 0 <= n <= 5 * 106
        
        int count = 0;
        for (int i = 2; i < n; i++) {
            if (isPrime(i)) {
                count++;
            }
        }
        return count;
    }
    
    public boolean isPrime(int n) {
        //check if n
        if (n <= 1) {
            return false;
        } else if (n <= 3) {
            return true;
        } else if (n % 2 == 0 || n % 3 == 0) {
            return false;
        } else {
            for (int i = 5; i * i <= n; i += 6) {
                if (n % i == 0 || n % (i + 2) == 0) {
                    return false;
                }
            }
            return true;
        }
    }
}

File: MutableAI/JavaScript/Q7.js
/**
 * @param {number} n
 * @return {number}
 */
var countPrimes = function(n) {
    //Given an integer n, return the number of prime numbers that are strictly less than n.

    //sieve of eratosthenes

    //create an array of all numbers from 2 to n
    //set all numbers to true except 2 and 3, which are prime.
    //for each number in the array, if it is true, then set all multiples of
    //to false.
    //count the number of true values in the array.
    //return the number of true values.
    let primes = [2,3];
    let count = 2
    for(let i = 4; i < n; i++){
        let isPrime = true;
        for(let j = 0; j < primes.length; j++){
            if(i % primes[j] === 0){
                isPrime = false;
                break;
            }
        }
        if(isPrime){
            primes.push(i);
            count++;
        }
    }
    return count;
}
